cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -1648,6 +1661,60 @@ String zkConnectionString(int connectToReplicas)  {
+                                acFut.compose(ac -> {
+                                    if (ac == null) {
+                                        kafkaPodsUpdatedDynamically.put(finalPodId, null);
+                                        log.warn("Could not create admin client.");
+                                        return Future.succeededFuture();
+                                    } else {
+                                        Future<Map<ConfigResource, Config>> futCurrent = kbch.getCurrentConfig(finalPodId, ac);
+                                        Future<ConfigMap> futDesired = configMapOperations.getAsync(namespace, KafkaCluster.metricAndLogConfigsName(name));
+                                        log.debug("Determining kafka pod {} dynamic update ability", finalPodId);
+                                        return CompositeFuture.join(futCurrent, futDesired)
+                                                .compose(res -> {
+                                                    KafkaBrokerConfigurationDiff configurationDiff = new KafkaBrokerConfigurationDiff(res.result().resultAt(0), res.result().resultAt(1), kafkaCluster.getKafkaVersion(), finalPodId);
+
+                                                    if (configurationDiff.getDiff().asOrderedProperties().asMap().size() > 0) {
+                                                        log.debug("kafka changed, dynamically? : {}", !configurationDiff.cannotBeUpdatedDynamically());
+                                                        kafkaPodsUpdatedDynamically.put(finalPodId, !configurationDiff.cannotBeUpdatedDynamically());
+                                                        AlterConfigsResult alterConfigResult = ac.incrementalAlterConfigs(configurationDiff.getUpdatedConfig(), new AlterConfigsOptions());
+                                                        for (Map.Entry entry : alterConfigResult.values().entrySet()) {
+                                                            KafkaFuture kafkaFuture = (KafkaFuture) entry.getValue();
+                                                            try {
+                                                                log.debug("AlterConfig result {}", kafkaFuture.get(operationTimeoutMs, TimeUnit.MILLISECONDS));
+                                                            } catch (InvalidRequestException | InterruptedException | ExecutionException | TimeoutException e) {
+                                                                log.warn("Error during dynamic reconfiguration. Rolling the pod. {}", e.getMessage());
+                                                                kafkaPodsUpdatedDynamically.put(finalPodId, false);
+                                                            }
+                                                        }
+                                                    } else {
+                                                        log.debug("kafka not changed");
```

Comment:
Again, weird logging statement. Not nicely written, not recomciliation -> to be fixed or removed.